src/cmn/timeoutManager.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * timeoutManager.cpp
+ *
+ *  Created on: Feb 16, 2020
+ *      Author: Anjana_Sreekumar
+ */
+
+#include "timeoutManager.h"
+#include "log.h"
+
+#include <cstring>
+#include <stdio.h>
+#include <sys/timerfd.h>
+#include <unistd.h>
+
+TimeoutManager::Timer::Timer() : fd_m(0)
+{
+    tspec_m.it_value.tv_sec = defaultTimeoutDurationSeconds_c;
+    tspec_m.it_value.tv_nsec = defaultTimeoutDurationNanoSeconds_c;
+    tspec_m.it_interval.tv_sec = defaultTimeoutDurationSeconds_c;
+    tspec_m.it_interval.tv_nsec = defaultTimeoutDurationNanoSeconds_c;
+
+    initTimer();
+}
+
+TimeoutManager::Timer::Timer(const struct itimerspec &tspec) : fd_m(0)
+{
+    memcpy(&tspec_m, &tspec, sizeof(struct itimerspec));
+
+    initTimer();
+}
+
+TimeoutManager::Timer::~Timer()
+{
+}
+
+void TimeoutManager::Timer::initTimer()
+{
+    fd_m = timerfd_create(CLOCK_MONOTONIC, 0);
+    if (fd_m <= 0)
+    {
+        log_msg(LOG_ERROR, "Failed to create timer\n");	    
+    }
+    else
+    {
+        log_msg(LOG_DEBUG, "Timer created successfully\n");
+    }	
+}
+
+bool TimeoutManager::Timer::startTimer()
+{
+    log_msg(LOG_DEBUG, "TimeoutManager::Timer::startTimer() ENTRY\n");    
+
+    bool rc = false;
+
+    if (fd_m != 0)
+    {
+        if (timerfd_settime(fd_m, 0, &tspec_m, NULL) == -1)
+        {
+            log_msg(LOG_ERROR, "Failed to set timeout duration\n");
+
+            rc = false;
+
+            // thread abort.
+        }
+    }
+
+    log_msg(LOG_DEBUG,"TimeoutManager::Timer::startTimer()-EXIT\n");
+
+    return rc;
+}
+
+bool TimeoutManager::Timer::stopTimer()
+{
```

Comment:
I dont see these functions used...Does  setting timeout 0 means no event on fd ?

Reply:
It stops(disarms) the timer, fd will still be valid..